google/cloud/testing_util/validate_metadata.cc

```diff
@@ -87,14 +87,11 @@ RoutingHeaders ExtractMDFromHeader(std::string header) {
  * `GTEST_USES_POSIX_RE`.
  */
 MATCHER_P(MatchesGlob, glob, "matches the glob: \"" + glob + "\"") {
-  // Translate the glob into a regex pattern.
-  auto matcher1 = absl::StrReplaceAll(glob, {{"*", "[^/]+"}});
-  // Create a second matcher that replaces all "/" with "%2F" to match the
+  // Create a matcher that replaces all "/" with "%2F" to match the
   // character replacement in `internal::UrlEncode`.
   auto decoded_glob = absl::StrReplaceAll(glob, {{"/", "%2F"}});
-  auto matcher2 = absl::StrReplaceAll(decoded_glob, {{"*", "[^/]+"}});
-  auto regex_string = absl::StrCat(matcher1, "|", matcher2);
-  std::regex regex(regex_string);
+  auto matcher = absl::StrReplaceAll(decoded_glob, {{"*", "[^/]+"}});
```

Comment:
For what it's worth, `absl::StrReplaceAll()` takes a list of replacements, so you could do it all in one call.
```
  auto matcher = absl::StrReplaceAll(glob, {{"/", "%2F"}, {"*", "[^/]+"}});
```